src/tcgwars/logic/impl/gen7/TeamUp.groovy

```diff
@@ -3076,11 +3082,11 @@ public enum TeamUp implements LogicCardInfo {
               damage 20
               afterDamage{
                 if(opp.hand.size() >= 4){
-                  if(opp.hand.size() == 4){
-                    opp.hand.showToMe("Your opponent's hand")
-                  }
-                  else{
-                    opp.hand.select(count:opp.hand.size() - 4,"Choose the cards to discard").discard()
+                  def randomOppHand = opp.hand.shuffledCopy()
+                  if (opp.hand.size() == 4){
+                    randomOppHand.shuffledCopy().showToMe("Your opponent's hand.")
```

Comment:
is the second `shuffledCopy()` operation needed here?

Reply:
Nope, leftover from before starting to use a variable. Nice catch 👍